Source/Core/base.bpl

```diff
@@ -217,8 +217,8 @@ function Set_Intersection<T>(a: Set T, b: Set T): Set T
   Set(MapAnd(a->val, b->val))
 }
 
-function Set_Choose<T>(a: Set T): T;
-axiom (forall<T> a: Set T :: {Set_Choose(a)} a == Set_Empty() || Set_Contains(a, Set_Choose(a)));
+function Choice<T>(a: [T]bool): T;
+axiom (forall<T> a: [T]bool :: {Choice(a)} a == MapConst(false) || a[Choice(a)]);
```

Comment:
Why ditch the `Set_` prefix and working with the `Set` wrapper type?

Would it be of any use to make the return type `Option<T>` and return `None` for empty sets instead of an unconstrained `T`?

Reply:
The earlier Set_Choose was defined on the type ```Set T``` of finite sets. In the ticket sample, I want to make a choice on the domain of an Lset which is ```[T]bool```. So I thought I would generalize Set_Choose to Choice allowing the idea to work on any set, finite or infinite.

> Would it be of any use to make the return type Option<T> and return None for empty sets instead of an unconstrained T?

Currently, I don't see any advantage.

Reply:
Oh, I did not realize that `Set` is for finite sets. Otherwise the domain of an `Lset` could have been a `Set`.